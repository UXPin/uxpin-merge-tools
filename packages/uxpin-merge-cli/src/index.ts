import 'source-map-support/register';
export { runProgram } from './program/runProgram';
export { Command } from './program/command/Command';
export { getToolVersion } from './program/utils/version/getToolVersion';
export { buildDesignSystemWithStorybook } from './steps/building/buildDesignSystemWithStorybook';
