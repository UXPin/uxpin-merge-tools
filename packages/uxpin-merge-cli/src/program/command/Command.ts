export enum Command {
  DUMP = 'dump',
  EXPERIMENT = 'experiment',
  INIT = 'init',
  GENERATE_PRESETS = 'generate-presets',
  PUSH = 'push',
  DELETE_VERSION = 'delete-version',
  SERVER = 'server',
  SUMMARY = 'summary',
}

export const DEFAULT_COMMAND: Command = Command.EXPERIMENT;
