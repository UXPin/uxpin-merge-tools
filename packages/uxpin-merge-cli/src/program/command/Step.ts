import { DSMetadata } from '../DSMeta';

export interface Step {
  shouldRun: boolean;
  exec: (infos: DSMetadata) => any;
}

export type StepExecutor = (designSystem: DSMetadata) => Promise<DSMetadata>;

export type StepWithoutDSExecutor = () => Promise<void>;
