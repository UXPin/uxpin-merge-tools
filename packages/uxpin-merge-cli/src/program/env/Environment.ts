export enum Environment {
  DEVELOPMENT = 'development',
  TEST = 'test',
  PRODUCTION = 'production',
}

export const DEFAULT_ENVIRONMENT: Environment = Environment.PRODUCTION;
