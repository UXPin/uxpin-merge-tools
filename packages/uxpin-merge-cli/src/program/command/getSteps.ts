import { ProgramArgs } from '../args/ProgramArgs';
import { Command } from './Command';
import { getDeleteVersionCommandSteps } from './delete_version/getDeleteVersionCommandSteps';
import { getDumpCommandSteps } from './dump/getDumpCommandSteps';
import { getExperimentationCommandSteps } from './experimentation/getExperimentationCommandSteps';
import { getGeneratePresetsCommandSteps } from './generate_presets/getGeneratePresetsCommandSteps';
import { getInitCommandSteps } from './init/getInitCommandSteps';
import { getPushCommandSteps } from './push/getPushCommandSteps';
import { getServerCommandSteps } from './server/getServerCommandSteps';
import { Step } from './Step';
import { getSummaryCommandSteps } from './summary/getSummaryCommandSteps';

export function getSteps(args: ProgramArgs): Step[] {
  switch (args.command) {
    case Command.DUMP:
      return getDumpCommandSteps();

    case Command.EXPERIMENT:
      return getExperimentationCommandSteps(args);

    case Command.INIT:
      return getInitCommandSteps(args);

    case Command.GENERATE_PRESETS:
      return getGeneratePresetsCommandSteps(args);

    case Command.PUSH:
      return getPushCommandSteps(args);

    case Command.DELETE_VERSION:
      return getDeleteVersionCommandSteps(args);

    case Command.SERVER:
      return getServerCommandSteps(args);

    case Command.SUMMARY:
      return getSummaryCommandSteps();

    default:
      return [];
  }
}
