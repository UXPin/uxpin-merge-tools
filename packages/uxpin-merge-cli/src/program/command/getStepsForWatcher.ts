import { ProgramArgs } from '../args/ProgramArgs';
import { Command } from './Command';
import { getExperimentationWatchCommandSteps } from './experimentation/getExperimentationWatchCommandSteps';
import { Step } from './Step';

export function getStepsForWatcher(args: ProgramArgs): Step[] {
  if (args.command === Command.EXPERIMENT) {
    return getExperimentationWatchCommandSteps(args);
  }

  return [];
}
