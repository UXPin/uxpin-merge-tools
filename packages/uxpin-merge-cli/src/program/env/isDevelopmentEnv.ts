import { Environment } from './Environment';

export function isDevelopmentEnv(): boolean {
  return process.env.UXPIN_ENV === Environment.DEVELOPMENT;
}
