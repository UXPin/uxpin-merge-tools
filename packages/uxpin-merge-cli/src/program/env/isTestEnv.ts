import { Environment } from './Environment';

export function isTestEnv(): boolean {
  return process.env.UXPIN_ENV === Environment.TEST;
}
