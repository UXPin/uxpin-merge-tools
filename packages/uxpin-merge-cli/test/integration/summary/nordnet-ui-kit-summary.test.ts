import { Command } from '../../../src';
import { Environment } from '../../../src/program/env/Environment';
import { nordnetUiKitSummaryStub } from '../../resources/stubs/nordnetUiKit';
import { runUXPinMergeCommand } from '../../utils/command/runUXPinMergeCommand';
import { setTimeoutBeforeAll } from '../../utils/command/setTimeoutBeforeAll';
import { setupStubbyServer } from '../../utils/stubby/setupStubbyServer';

const CURRENT_TIMEOUT = 300000;

setTimeoutBeforeAll(CURRENT_TIMEOUT);

describe('summary command integration', () => {
  const { getTlsPort } = setupStubbyServer(nordnetUiKitSummaryStub);

  describe('summary command prints ', () => {
    it('prints the list of components found in nordnet-ui-kit example', () => {
      // when
      return runUXPinMergeCommand({
        cwd: 'resources/repos/nordnet-ui-kit',
        env: {
          UXPIN_API_DOMAIN: `0.0.0.0:${getTlsPort()}`,
          UXPIN_ENV: Environment.TEST,
        },
        params: [Command.SUMMARY, '--config="../../configs/nordnet-ui-kit-uxpin.config.js"'],
      }).then((output) => {
        // then
        expect(output).toMatchSnapshot();
      });
    });
  });
});
