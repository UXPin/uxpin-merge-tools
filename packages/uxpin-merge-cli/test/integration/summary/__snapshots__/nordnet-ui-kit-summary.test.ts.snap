// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`summary command integration summary command prints  prints the list of components found in nordnet-ui-kit example 1`] = `
"[1mUncategorized[22m

    [1mAlert[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mAnimate[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mAvatar[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mBadge[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mButton[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mDropdown[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mFlag[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mIcon[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mInput[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mLabeledValue[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mLi[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mLogo[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mPane[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mRadioGroup[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mSegmentedControl[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mSpinner[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mTable[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mTbody[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mTd[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mTfoot[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mTh[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mThead[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mTooltip[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mTr[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mUl[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

[33mwarning [39mCannot serialize component properties in:
[34msrc/components/alert/alert.jsx[39m

[33mwarning [39mCannot serialize component properties in:
[34msrc/components/animate/animate.jsx[39m

[33mwarning [39mCannot serialize component properties in:
[34msrc/components/avatar/avatar.jsx[39m

[33mwarning [39mCannot serialize component properties in:
[34msrc/components/badge/badge.jsx[39m

[33mwarning [39mCannot serialize component properties in:
[34msrc/components/button/button.jsx[39m

[33mwarning [39mCannot serialize component properties in:
[34msrc/components/dropdown/dropdown.jsx[39m

[33mwarning [39mCannot parse type of a property 'aria-hidden' in:
[34msrc/components/icon/icon.jsx[39m

[33mwarning [39mCannot parse type of a property 'focusable' in:
[34msrc/components/icon/icon.jsx[39m

[33mwarning [39mCannot serialize component properties in:
[34msrc/components/labeled-value/labeled-value.jsx[39m

[33mwarning [39mCannot serialize component properties in:
[34msrc/components/li/li.jsx[39m

[33mwarning [39mCannot serialize component properties in:
[34msrc/components/pane/pane.jsx[39m

[33mwarning [39mCannot serialize component properties in:
[34msrc/components/spinner/spinner.jsx[39m

[33mwarning [39mCannot serialize component properties in:
[34msrc/components/td/td.jsx[39m

[33mwarning [39mCannot serialize component properties in:
[34msrc/components/th/th.jsx[39m

[33mwarning [39mCannot serialize component properties in:
[34msrc/components/tooltip/tooltip.jsx[39m

"
`;
